filters/f_hwtransfer.c

```diff
@@ -129,6 +130,14 @@ static bool select_format(struct priv *p, int input_fmt,
     int *upload_fmts = &p->upload_fmts[p->fmt_upload_index[index]];
     int num_upload_fmts = p->fmt_upload_num[index];
 
+    if (p->ctx->try_upload) {
+        upload_fmts = talloc_array_ptrtype(NULL, upload_fmts, num_upload_fmts);
+        memcpy(upload_fmts, &p->upload_fmts[p->fmt_upload_index[index]], num_upload_fmts * sizeof(int));
+    }
+
+try_next:; // Try only upload formats, if all of them fail, give up. We could attempt
```

Comment:
Can't this just be a `do {} while()` loop?

Reply:
changed to while loop